pkg/proxy/roundrobin.go

```diff
@@ -227,7 +228,8 @@ func (lb *LoadBalancerRR) OnUpdate(allEndpoints []api.Endpoints) {
 
 	// Update endpoints for services.
 	for _, svcEndpoints := range allEndpoints {
-		key := balancerKey(svcEndpoints.Name)
+		name := types.NewNamespacedName(svcEndpoints.Namespace, svcEndpoints.Name)
+		key := balancerKey(*name)
```

Comment:
Agreed, I'll do that.  I wanted to get a first-pass review before rewriting your recently written code :-)